### ISSUE TITLE
kustomize-controller reports Kyverno ClusterPolicy "configured" on every reconciliation

### ISSUE THREAD

**Issue by balonik** [fluxcd/flux2#2620]
### Describe the bug

Every time that Flux's kustomize-controller reconciles Kyverno `ClusterPolicy` resources are "configured" while all the other resources covered by the same Kustomization remain "unchanged".

This seems to affect only Kyverno `ClusterPolicy` resources which has `background: true`, which is the default. Might be a coincidence, but I have 5 policies where 1 has `background: false` and this one is not reported as "configured" while the other 4 are.

This is the diff between one of the ClusterPolicy being affected:
```
$ diff before after 
11c11
<   generation: 1026
---
>   generation: 1040
16c16
<   resourceVersion: "864110"
---
>   resourceVersion: "865689"
```

Neither field is specified in the manifest files.

### Steps to reproduce

1. Install Flux 2 v0.28.5
2. Deploy Kyverno - I have tested with `1.6.2` and `1.5.8` just in case.
3. Deploy any Kyverno `ClusterPolicy` resource
4. Observe that every time Flux reconciles this object, it notes that it "configured" it, changing its resource version and generation.

### Expected behavior

ClusterPolicies are not reconfigured everytime by Flux.

### OS / Distro

N/A

### Flux version

N/A

### Flux check

N/A

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by 0ur0b0r0s** [comment]
### Facing the same issue
Having the same problem here, but with a different resource, in my case, it's Kibana. Recently, I had to realocate my Elasticsearch stack (elasticsearch, kibana, filebeat, heartbeat), to use a node from a different nodepool. After I commited my changes to git, realocating all the resources of my stack to the other node, flux reconfigured all these resources, including Kibana. Just as expected, but somehow, after that, every time that Flux's kustomize-controller reconciles Kibana ClusterPolicy resources are "configured" while all the other resources covered by the same Kustomization remain "unchanged". Just as you reported before @balonik.

### Screenshots and recordings
![image](https://user-images.githubusercontent.com/38080850/165829248-bc6e1159-7c18-4f2d-9904-2070b7b0cce0.png)

### Flux version
flux version 0.23.0

**Comment by seh** [comment]
This may be related to kyverno/kyverno#2711, for which there's a fix in the works.

**Comment by balonik** [comment]
Yes, my issue is related to the linked kyverno issue, where kyverno updates `spec` of the Policy and Flux updates it back, in never ending circle.

**Comment by stefanprodan** [comment]
I've reached out the Kyverno team and they said this will be fixed in their next minor version. Kyverno will not update the `spec`, instead it will update the `status` sub-resource, which Flux expects from other controllers.

**Comment by stefanprodan** [comment]
This was fixed in Kyverno  1.7.0
